cppsdk/include/playfab/PlayFabEvent.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#ifndef DISABLE_PLAYFABENTITY_API
+
+#include <playfab/PlayFabError.h>
+#include <playfab/PlayFabEventsDataModels.h>
+
+namespace PlayFab
+{
+    enum struct PlayFabEventServerProcessingType
+    {
+        Default,
+        Lightweight,
+        Heavyweight
+    };
+
+    /// <summary>
+    /// Interface for any event
+    /// </summary>
+    struct IPlayFabEvent
+    {
+    public:
+        virtual ~IPlayFabEvent() {}
+    };
+
+    /// <summary>
+    /// PlayFab-specific implementation of an event
+    /// </summary>
+    struct PlayFabEvent : public IPlayFabEvent
+    {
+    public:
+        PlayFabEventServerProcessingType serverProcessingType;
+        EventsModels::EventContents eventContents;
```

Comment:
>EventsModels::EventContents [](start = 8, length = 27)

As we discussed, let's specialize this and restrict to known types for now with setters exposed.

Reply:
Will be replaced in the next iteration

---
In reply to: [227944135](https://github.com/PlayFab/XPlatCppSdk/pull/6#discussion_r227944135) [](ancestors = 227944135)